platforms/shared/charts/vault-script/scripts/bevel-vault.sh

```diff
@@ -0,0 +1,184 @@
+    aws configure set aws_access_key_id $CLIENT_ID
+    aws configure set aws_secret_access_key $CLIENT_SECRET
+    aws configure set default.region "eu-west-1"
+}
+
+function readAWSSecretManager {
+    RESP=$(aws secretsmanager get-secret-value --secret-id $VAULT_SECRET_PATH/${1} | jq -r ".SecretString")
+    if [[ "$RESP" == "" ]]
+    then
+        VAULT_SECRET="null"
+    else
+        VAULT_SECRET=$(echo ${RESP} | sed -e 's!\\n !\\n!g') # Remove the extra space in the new lines of certificate
+    fi
+}
+
+#Arg1: Secret keyname
+#Arg2: Secret value
+function writeAWSSecretManger {
+    # Format data as per required by AKV
+    VAULT_RESPONSE=$(aws secretsmanager create-secret --name $VAULT_SECRET_PATH/${1} --secret-string file://${2})
+}
+
+##############################################################################################################################
+################################################## AZURE key vault functions starts ##########################################
+##############################################################################################################################
+
+function initAzureVaultToken {
+    VAULT_TOKEN=$(curl --location --request POST https://login.microsoftonline.com/${TENANT_ID}/oauth2/v2.0/token \
+                                --form 'grant_type="client_credentials"' \
+                                --form 'client_id="'${CLIENT_ID}'"' \
+                                --form 'client_secret="'${CLIENT_SECRET}'"' \
+                                --form 'scope="https://vault.azure.net/.default"' | jq '.access_token' | sed -e 's/^"//' -e 's/"$//') # sed to remove the quotes at from start and end
+}
+
+#Arg1: Secret keyname
+function readAzureVaultSecret {
+    VAULT_SECRET=$(curl --location --request GET ${VAULT_ADDR}/secrets/${1}?api-version=7.3 \
+                                    --header 'Authorization: Bearer '${VAULT_TOKEN} | jq '.value' | sed -e 's/^"//' -e 's/"$//') # sed to remove the quotes at from start and end
+}
+
+#Arg1: Secret keyname
+#Arg2: Secret value
+function writeAzureVaultSecret {
+    # Format data as per required by AKV
+    JSON_PAYLOAD=$(cat ${2} | sed -e 's!"!\\"!g')
+    VAULT_RESPONSE=$(curl --location --request PUT ${VAULT_ADDR}/secrets/${1}?api-version=7.3 \
+                                    --header 'Authorization: Bearer '${VAULT_TOKEN} --header 'Content-Type: application/json' \
+                                    --data-raw '{"value": "'"$JSON_PAYLOAD"'"}' )
+}
+
+
+##############################################################################################################################
+################################################## Vault main handler function ###############################################
+##############################################################################################################################
+
+vaultBevelFunc() {
+    if [[ $VAULT_TYPE = "hashicorp" ]]; then
```

Comment:
@sownak the vault conditions are determined here based on the vault type passed into container env variable.